dbus_service.py

```diff
@@ -539,61 +557,99 @@ def sign_of_life(self):
-        4. If not in dry run mode, sets the DBus values.
-        5. Updates the index.
-        6. Handles various exceptions that may occur during the update process:
-            - requests.exceptions.RequestException: Logs an HTTP error if the last update was successful.
-            - ValueError: Logs a value error if the last update was successful.
-            - Exception: Logs a general error if the last update was successful.
-        7. Logs a recovery message if the update was successful after a previous failure.
-
-        Attributes:
-            successful (bool): Indicates whether the update was successful.
+        Updates inverter data from the DTU (Data Transfer Unit) and sets DBus values if the data is up-to-date.
+
+        Main logic:
+        - In timeout mode: Always attempt reconnect every RetryAfterSeconds. Only set zero values after ErrorStateAfterSeconds has elapsed since last success.
+        - In retrycount mode: After min_retries_until_fail failures, wait RetryAfterSeconds before next attempt and set zero values immediately.
+        - Always updates the DBus update index after a refresh.
+        - Tracks success/failure state and manages reconnect timing.
+
+        Exception handling:
+        - Catches and logs HTTP, value, and general exceptions during update.
+        - Ensures update state is finalized regardless of outcome.
+
+        Returns:
+            None
         """
         logging.debug("_update")
         successful = False
+        now = time.time()
         try:
-            # update data from DTU once per _update call:
-            self._refresh_data()
-
-            if self.is_data_up2date():
-                if self.dry_run:
-                    logging.info("DRY RUN. No data is sent!!")
-                else:
-                    self.set_dbus_values()
-            self._update_index()
-            successful = True
+            if self.error_mode == "timeout" and self.error_state_after_seconds > 0:
+                # Set zero values only after ErrorStateAfterSeconds has elapsed since last success
+                if (not self.last_update_successful and (now - self._last_update) >= self.error_state_after_seconds):
+                    self._handle_reconnect_wait()
+                # Always allow a reconnect attempt every RetryAfterSeconds
+                if (now - self._last_update) >= self.retry_after_seconds:
+                    successful = self._refresh_and_update()
+                # In normal operation (no error), always call _refresh_data on every update
+                if self.last_update_successful:
+                    successful = self._refresh_and_update()
+            elif self.error_mode == "retrycount":
+                # Classic retry-count-based error handling
+                if self.failed_update_count >= self.min_retries_until_fail:
+                    self._handle_reconnect_wait()
+                if self._should_refresh_data(now):
```

Comment:
Ich würde statt dieser Methode eine oder mehrere lokale Variablen mit sprechenden Namen bevorzugen. Dies gehört eng zur Logik hier und sollte m.E. NICHT versteckt werden, weil man hier doch immer nachschauen wird, dass in der Methode passiert. Die Var. sind dann aber sprechend und machen es lesbarer.